### PR TITLE
🌱 align flags with CAPI

### LINKED ISSUE #2072: Make CAPV command line flags consistent with core CAPI
I diffed main.go files and the main difference is that CAPV is missing some flags.

Flags missing in CAPV:
fs.DurationVar(&leaderElectionLeaseDuration, "leader-elect-lease-duration", 15*time.Second,
"Interval at which non-leader candidates will wait to force acquire leadership (duration string)")

fs.DurationVar(&leaderElectionRenewDeadline, "leader-elect-renew-deadline", 10*time.Second,
"Duration that the leading controller manager will retry refreshing leadership before giving up (duration string)")

fs.DurationVar(&leaderElectionRetryPeriod, "leader-elect-retry-period", 2*time.Second,
"Duration the LeaderElector clients should wait between tries of actions (duration string)")

fs.BoolVar(&enableContentionProfiling, "contention-profiling", false,
"Enable block profiling, if profiler-address is set.")

fs.Float32Var(&restConfigQPS, "kube-api-qps", 20,
"Maximum queries per second from the controller client to the Kubernetes API server. Defaults to 20")

fs.IntVar(&restConfigBurst, "kube-api-burst", 30,
"Maximum number of queries that should be allowed in one burst from the controller client to the Kubernetes API server. Default 30")

fs.StringVar(&webhookCertDir, "webhook-cert-dir", "/tmp/k8s-webhook-server/serving-certs/",
"Webhook cert dir, only used when webhook-port is specified.")

Additionally we should start using the profiler server built into CR since CR v0.17 instead of starting our own (cf with CAPI main.go)

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

Aligns flags to the ones used in CAPI:

- adds the flag `--leader-elect-lease-duration`
- adds the flag `--leader-elect-renew-deadline`
- adds the flag `--leader-elect-retry-period`
- adds the flag `--contention-profiling`
- adds the flag `--kube-api-qps`
- adds the flag `--kube-api-burst`
- adds the flag `--webhook-cert-dir`
- replaces the profiler webserver by the controller-runtime provided one in `main.go`

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2072

**Special notes for your reviewer**:

Detailed diff in help output:

```diff
0a1
>       --contention-profiling                     Enable block profiling, if profiler-address is set.
8a10,11
>       --kube-api-burst int                       Maximum number of queries that should be allowed in one burst from the controller client to the Kubernetes API server. Default 30 (default 30)
>       --kube-api-qps float32                     Maximum queries per second from the controller client to the Kubernetes API server. Defaults to 20 (default 20)
10a14,16
>       --leader-elect-lease-duration duration     Interval at which non-leader candidates will wait to force acquire leadership (duration string) (default 15s)
>       --leader-elect-renew-deadline duration     Duration that the leading controller manager will retry refreshing leadership before giving up (duration string) (default 10s)
>       --leader-elect-retry-period duration       Duration the LeaderElector clients should wait between tries of actions (duration string) (default 2s)
16d21
22c27,29
<       --sync-period duration                     The interval at which cluster-api objects are synchronized (default 10m0s)
---
>       --sync-period duration                     The minimum interval at which watched resources are reconciled (e.g. 15m) (default 10m0s)
29a37,43
>       --webhook-cert-dir string                  Webhook cert dir, only used when webhook-port is specified. (default "/tmp/k8s-webhook-server/serving-certs/")
```
_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
Align flags to core CAPI.
```

### CODE REVIEW

File: controllers/vspherecluster_controller.go [hunk 1]
```diff
@@ -97,11 +97,12 @@ func AddClusterControllerToManager(ctx *context.ControllerManagerContext, mgr ma
 		return ctrl.NewControllerManagedBy(mgr).
 			Named(controllerNameShort).
 			For(clusterControlledType).
+			WithOptions(options).
 			Watches(
 				&vmwarev1.VSphereMachine{},
 				handler.EnqueueRequestsFromMapFunc(reconciler.VSphereMachineToCluster),
 			).
-			WithOptions(controller.Options{MaxConcurrentReconciles: ctx.MaxConcurrentReconciles}).
+			WithEventFilter(predicates.ResourceNotPausedAndHasFilterLabel(ctrl.LoggerFrom(ctx), ctx.WatchFilterValue)).
```
Comment (chrischdi) [inline controllers/vspherecluster_controller.go:105]: @sbueringer : I think filtering out paused objects via predicates should be good?!

**Comment by chrischdi** [comment]
This PR grew a lot. If its too much for folks I can split it up in separate ones.

**Comment by chrischdi** [comment]
Splitted this PR into multiple

File: main.go [hunk 2]
```diff
@@ -73,86 +81,99 @@ var (
...
-		"profiler-address",
-		defaultProfilerAddr,
-		"Bind address to expose the pprof profiler (e.g. localhost:6060)")
-	flag.DurationVar(
-		&syncPeriod,
-		"sync-period",
-		defaultSyncPeriod,
-		"The interval at which cluster-api objects are synchronized")
+
 	flag.IntVar(
```
Comment (sbueringer) [inline main.go:94]: we moved all other vars to use fs. this one is still using flag (?)

File: main.go [hunk 3]
```diff
@@ -73,86 +81,99 @@ var (
...
+		"The minimum interval at which watched resources are reconciled (e.g. 15m)")
+
+	fs.Float32Var(&restConfigQPS, "kube-api-qps", 20,
+		"Maximum queries per second from the controller client to the Kubernetes API server. Defaults to 20")
+
+	fs.IntVar(&restConfigBurst, "kube-api-burst", 30,
+		"Maximum number of queries that should be allowed in one burst from the controller client to the Kubernetes API server. Default 30")
+
+	fs.IntVar(&webhookOpts.Port, "webhook-port", defaultWebhookPort,
+		"Webhook Server port (set to 0 to disable)")
```
Comment (sbueringer) [inline main.go:168]:
```suggestion
	fs.IntVar(&webhookOpts.Port, "webhook-port", defaultWebhookPort,
		"Webhook Server port")
```

0 just means use the default port (9443)

File: main.go [hunk 4]
```diff
@@ -54,13 +54,21 @@ import (
...
-	setupLog   = ctrl.Log.WithName("entrypoint")
-	logOptions = logs.NewOptions()
-
-	managerOpts     manager.Options
-	webhookOpts     webhook.Options
-	syncPeriod      time.Duration
-	profilerAddress string
+	setupLog       = ctrl.Log.WithName("entrypoint")
+	logOptions     = logs.NewOptions()
+	controllerName = "cluster-api-provider-vsphere-manager"
```
Comment (sbueringer) [inline main.go:59]:
```suggestion
	controllerName = "cluster-api-vsphere-manager"
```
maybe. Not sure if there is any consistency for that stuff

File: main.go (hunk 2 above)
Comment (chrischdi) [reply main.go:94]: jup, missed that one 👍

**Comment by sbueringer** [comment]
/lgtm
Thx!

**Pull request merged**